app/src/main/java/com/w3engineers/unicef/util/helper/InAppShareUtil.java

```diff
@@ -0,0 +1,239 @@
+
+                for (int x = 0; x < bitMatrixWidth; x++) {
+                    pixels[offset + x] = bitMatrix.get(x, y) ?
+                            context.getResources().getColor(R.color.colorPrimaryDark) :
+                            context.getResources().getColor(R.color.white);
+                }
+            }
+            urlQrBitmap = Bitmap.createBitmap(bitMatrixWidth, bitMatrixHeight, Bitmap.Config.ARGB_4444);
+            urlQrBitmap.setPixels(pixels, 0, 150, 0, 0, bitMatrixWidth, bitMatrixHeight);
+
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Nullable
+    public String serverInit() {
+
+        String myAddress = getMyIpAddress();
+        int httpPort = random.nextInt(9000) + 1000;
+
+        serverAddress = "http://" + myAddress + ":" + httpPort;
+
+        String backApkPath = backupApkAndGetPath();
+
+        InstantServer.getInstance().setPort(httpPort).setFilePath(backApkPath);
+
+        return serverAddress;
+    }
+
+    @Nullable
+    public String backupApkAndGetPath() {
+
+        Context context = App.getContext();
+
+        Intent mainIntent = new Intent(Intent.ACTION_MAIN, null);
+        mainIntent.addCategory(Intent.CATEGORY_LAUNCHER);
+
+        List pkgAppsList = context.getPackageManager().queryIntentActivities(mainIntent, 0);
+        String myApplicationPackageName = context.getPackageName();
+        String myApplicationName = context.getResources().getString(R.string.app_name);
+        String backupFolder = ".backup";
+
+        for (Object object : pkgAppsList) {
+
+            ResolveInfo resolveInfo = (ResolveInfo) object;
+            File appFile = new File(resolveInfo.activityInfo.applicationInfo.publicSourceDir);
+
+            try {
+
+                String file_name = resolveInfo.loadLabel(context.getPackageManager()).toString();
+
+                if (file_name.equalsIgnoreCase(myApplicationName) &&
+                        appFile.toString().contains(myApplicationPackageName)) {
+
+                    File file = new File(Environment.getExternalStorageDirectory().toString() + "/" +
+                            context.getString(R.string.app_name));
+                    file.mkdirs();
+                    // Preparing a backup apk folder and it is hidden
+                    File backUpFolder = new File(file.getAbsolutePath() + "/" + backupFolder);
+                    backUpFolder.mkdirs();
+                    backUpFolder = new File(backUpFolder.getPath() + "/" + file_name + ".apk");
+                    backUpFolder.createNewFile();
+
+                    InputStream in = new FileInputStream(appFile);
+                    OutputStream out = new FileOutputStream(backUpFolder);
+
+                    byte[] buf = new byte[1024];
+                    int len;
+                    while ((len = in.read(buf)) > 0) {
+                        out.write(buf, 0, len);
+                    }
+                    in.close();
```

Comment:
if we use try-with-resource which is a good java 8 feature then we don't need to bother about closing this Input stream and output stream

Reply:
Thanks for sharing but this xml file is not exist in current codebase.